src/main/java/net/earthcomputer/clientcommands/command/MinesweeperCommand.java

```diff
@@ -0,0 +1,341 @@
+        this.minesLeft = mines;
+        this.emptyTilesRemaining = width * height - mines;
+    }
+
+    @Override
+    protected void init() {
+        this.topLeftX = (this.width - gameWidth) / 2;
+        this.topLeftY = (this.height - gameHeight) / 2;
+    }
+
+    @Override
+    public void render(GuiGraphics graphics, int mouseX, int mouseY, float tickDelta) {
+        renderBackground(graphics, mouseX, mouseY, tickDelta);
+
+        graphics.drawString(minecraft.font, "Mines Left: " + minesLeft, topLeftX, topLeftY - 10, 0xff_ffffff);
+        graphics.drawCenteredString(minecraft.font, title.getString(), topLeftX + gameWidth / 2, topLeftY - 20, 0xff_ffffff);
+        {
+            String str = "Time Played: " + Math.ceilDiv(ticksPlaying, 20) + "s";
+            graphics.drawString(minecraft.font, str, topLeftX + gameWidth - minecraft.font.width(str), topLeftY - 10, deathCoords != null ? ChatFormatting.RED.getColor() : (emptyTilesRemaining == 0 ?  ChatFormatting.GREEN.getColor() : ChatFormatting.WHITE.getColor()));
+        }
+
+        graphics.blitSprite(TOP_LEFT, topLeftX, topLeftY, 12, 12);
+        for (int i = 0; i < boardWidth; i++) graphics.blitSprite(TOP, topLeftX + 12 + i * 16, topLeftY, 16, 12);
+        graphics.blitSprite(TOP_RIGHT, topLeftX + 12 + boardWidth * 16, topLeftY, 8, 12);
+        for (int i = 0; i < boardHeight; i++) graphics.blitSprite(LEFT, topLeftX, topLeftY + 12 + i * 16, 12, 16);
+        for (int i = 0; i < boardHeight; i++) graphics.blitSprite(RIGHT, topLeftX + 12 + boardWidth * 16, topLeftY + 12 + i * 16, 8, 16);
+        graphics.blitSprite(BOTTOM_LEFT, topLeftX, topLeftY + 12 + boardHeight * 16, 12, 8);
+        for (int i = 0; i < boardWidth; i++) graphics.blitSprite(BOTTOM, topLeftX + 12 + i * 16, topLeftY + 12 + boardHeight * 16, 16, 8);
+        graphics.blitSprite(BOTTOM_RIGHT, topLeftX + 12 + boardWidth * 16, topLeftY + 12 + boardHeight * 16, 8, 8);
+
+        for (int x = 0; x < boardWidth; x++) {
+            for (int y = 0; y < boardHeight; y++) {
+                boolean hovered = Mth.floorDiv(mouseX - topLeftX - 12, 16) == x && Mth.floorDiv(mouseY - topLeftY - 12, 16) == y;
+                graphics.blitSprite(getTileSprite(x, y, hovered), topLeftX + x * 16 + 12, topLeftY + y * 16 + 12, 16, 16);
+            }
+        }
+    }
+
+    @Override
+    public void tick() {
+        if (ticksPlaying > 0 && !gameFinished()) {
+            ticksPlaying += 1;
+        }
+    }
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+        dragging = button;
+        return false;
+    }
+
+    @Override
+    public boolean mouseReleased(double mouseX, double mouseY, int button) {
+        dragging = null;
+
+        int tileX = Mth.floorDiv((int) (mouseX - topLeftX - 12), 16);
+        int tileY = Mth.floorDiv((int) (mouseY - topLeftY - 12), 16);
+
+        if (isWithinBounds(tileX, tileY)) {
+            if (button == 0 && !gameFinished()) {
```

Comment:
Please use `GLFW` constants instead of magic numbers.

Reply:
Or `InputConstants`, rather.